scripts/module_list/README.md

```diff
@@ -0,0 +1,2 @@
+# Module list
+A script that generates a list (in markdown) of all available lmod modules.
```

Comment:
Add basic usage info + list of requirements (Lmod must be available, `$LMOD_CMD` must specify path to `lmod` binary, `$MODULEPATH` must be set up correctly, required Python packages outside of Python stdlib lib (see `requirements.txt`), etc.)

Reply:
README.md has been updated.